File: android/app/src/main/java/com/vxin/app/data/repository/AuthRepository.kt
package com.vxin.app.data.repository

import com.vxin.app.core.storage.AccountStore
import com.vxin.app.core.storage.TokenStore
import com.vxin.app.data.api.AuthApi
import com.vxin.app.data.model.Account
import com.vxin.app.data.model.LoginRequest
import com.vxin.app.data.model.RegisterRequest
import com.vxin.app.data.model.ResetPasswordRequest
import com.vxin.app.data.model.User
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class AuthRepository @Inject constructor(
    private val api: AuthApi,
    private val tokenStore: TokenStore,
    private val accountStore: AccountStore,
) {
    suspend fun login(phone: String, password: String): User {
        android.util.Log.d("LOGIN_DIAG", "LOGIN_REQUEST_OBJECT_CREATE_START")
        val request = LoginRequest(phone.trim(), password)
        android.util.Log.d("LOGIN_DIAG", "LOGIN_REQUEST_OBJECT_CREATED phone=${request.phone.take(3)}***")

        android.util.Log.d("LOGIN_DIAG", "LOGIN_RETROFIT_CALL_START")
        val res = api.login(request)
        android.util.Log.d("LOGIN_DIAG", "LOGIN_RETROFIT_CALL_SUCCESS token=${res.token.take(10)}... userId=${res.user.id}")

        applyAuth(res.token, res.user)
        return res.user
    }

    suspend fun register(phone: String, password: String, username: String, inviteCode: String): User {
        val res = api.register(RegisterRequest(phone.trim(), password, username.trim(), inviteCode.trim()))
        applyAuth(res.token, res.user)
        return res.user
    }

    private fun applyAuth(token: String, user: User) {
        tokenStore.token = token
        accountStore.upsertActive(Account(user.id, user.username, user.avatar, token))
    }

    suspend fun restoreSession(): User? {
        if (!tokenStore.isLoggedIn) return null
        return runCatching { api.me() }.getOrNull()
    }

    suspend fun logout() {
        runCatching { api.logout() }
        accountStore.activeId()?.let { accountStore.remove(it) }
        tokenStore.clear()
    }

    suspend fun resetPassword(phone: String, inviteCode: String, newPassword: String) {
        api.resetPassword(
            ResetPasswordRequest(
                phone = phone.trim(),
                inviteCode = inviteCode.trim(),
                newPassword = newPassword,
            ),
        )
    }
}

File: ios/Vxin/Data/Repositories/AuthRepository.swift
import Foundation

/// 认证仓库。与 Android AuthRepository 等价。
final class AuthRepository {
    static let shared = AuthRepository()
    private init() {}

    private let api = APIClient.shared

    func login(phone: String, password: String) async throws -> User {
        let res: AuthResponse = try await api.send(
            "api/auth/login", method: "POST",
            body: LoginBody(phone: phone.trimmingCharacters(in: .whitespaces), password: password),
            authorized: false
        )
        applyAuth(res)
        return res.user
    }

    private func applyAuth(_ res: AuthResponse) {
        KeychainStore.shared.token = res.token   // active token
        AccountStore.shared.upsertActive(StoredAccount(id: res.user.id, username: res.user.username, avatar: res.user.avatar, token: res.token))
    }

    func register(phone: String, password: String, username: String, inviteCode: String) async throws -> User {
        let res: AuthResponse = try await api.send(
            "api/auth/register", method: "POST",
            body: RegisterBody(
                phone: phone.trimmingCharacters(in: .whitespaces),
                password: password,
                username: username.trimmingCharacters(in: .whitespaces),
                inviteCode: inviteCode.trimmingCharacters(in: .whitespaces)
            ),
            authorized: false
        )
        applyAuth(res)
        return res.user
    }

    /// 找回密码(免登录):手机号+邀请码+新密码。与 web/Android 对齐。
    func resetPassword(phone: String, inviteCode: String, newPassword: String) async throws {
        let _: EmptyResponse = try await api.send(
            "api/auth/reset-password", method: "POST",
            body: ResetPasswordBody(
                phone: phone.trimmingCharacters(in: .whitespaces),
                inviteCode: inviteCode.trimmingCharacters(in: .whitespaces),
                newPassword: newPassword
            ),
            authorized: false
        )
    }

    /// 启动时凭已存 token 校验会话
    func restoreSession() async -> User? {
        guard KeychainStore.shared.isLoggedIn else { return nil }
        return try? await api.send("api/auth/me")
    }

    func logout() async {
        let _: EmptyResponse? = try? await api.send("api/auth/logout", method: "POST")
        if let active = AccountStore.shared.activeId() { AccountStore.shared.remove(active) }
        KeychainStore.shared.clear()
    }
}
